### ISSUE TITLE
Generate Documentation

### ISSUE THREAD

**Issue by paf31** [purescript/package-sets#13]
Since all packages in a set will build together, it makes sense to generate all documentation at once, so that we can show things like links to instances in other modules.

We should probably think about how this applies to Pursuit in the longer term if the `psc-package` experiment is a success 

@hdgarrood, any thoughts? I think we could potentially aim to have Pursuit build docs on a regular basis for each active package set, or something like that. Since we would build and cache all externs for a package set during CI anyway, we could use this to get a pretty great type search feature too.

**Comment by hdgarrood** [comment]
I feel like I've been out of the loop for quite a while so I'm not sure really how helpful I can be on this front, but this does sound very exciting to me; I'd certainly welcome work in this direction. In particular I can't think of any reasons not to at least attempt this.

**Comment by f-f** [comment]
All of this is being tracked in this Spago issue as well: https://github.com/spacchetti/spago/issues/89

I'll close this one as the linked issue has much more detail and an implementation plan (as this will need support from spago or psc-package and some compiler work too), so please refer to that one

**Comment by f-f** [comment]
I'm going to revive this issue to retarget it toward generating "Pursuit-like" documentation for every release of this repo.

Since [`spago-0.9.0`](https://github.com/spacchetti/spago/releases/tag/0.9.0.0) it is now possible to have a client-side search field added to the HTML docs generated by the compiler (thanks to the great work in [purescript-docs-search](https://github.com/spacchetti/purescript-docs-search)).
This means that we could generate static html documentation from CI for every release of this repo.

How I'd do it (all of this would happen on release CI):
- create an ephemeral spago project with all the packages in the set as dependencies (we can do this easily by downloading the release with `dhall-json` and templating a `spago.dhall` with the resulting list)
- run `spago docs`
- move the result to a folder with the release name
- commit it to the `gh-pages` branch

In this way we'd get the docs at something like `https://purescript.github.io/package-sets/whatever-release-name`

Some additional points:
- we might want some `index.html` to index all releases (much like the homepage for Stackage)
- we might want to not point to the GitHub Pages URL directly, but instead use something like `package-sets.purescript.org`, so that if the documentation starts taking too much disk space we can move off GitHub easily without breaking anyone's workflow

**Comment by f-f** [comment]
Update: @klntsky is working on this, you can find the work in progress at https://github.com/spacchetti/starsuit

**Comment by klntsky** [comment]
Yep, and the docs in their current state can be found [here](https://spacchetti.github.io/starsuit/html/index.html).

**Comment by f-f** [comment]
The [alpha for the new Registry](https://discourse.purescript.org/t/registry-alpha-launched/3146) has now launched, and this entails the automation of this repository - packages in the set will be managed through [the new API](https://github.com/purescript/registry#modify-the-package-set), and a new release will be cut once a day if there are any new updates.

We have been able to generate docs for a package set locally for quite some time now, but we plan to put these docs online (i.e. expanding what Starsuit is doing today). This work will likely be tracked in [the `registry-dev`](https://github.com/purescript/registry-dev) or [the `pursuit`](https://github.com/purescript/pursuit) repo.
